virtualizarr/tests/test_writers/test_icechunk.py

```diff
@@ -282,9 +290,25 @@ def test_write_loadable_variable(
     assert isinstance(pres_array, Array)
     assert pres_array.shape == (3, 4)
     assert pres_array.dtype == np.dtype("int32")
-    expected_ds = open_dataset(simple_netcdf4)
+    expected_ds = open_dataset(netcdf_path)
     expected_array = expected_ds["foo"].to_numpy()
     npt.assert_equal(pres_array, expected_array)
+    expected_ds.close()
+
+    # Now we can overwrite the simple_netcdf4 file with new data to make sure that
+    # the checksum_date is being used to determine if the data is valid
+    arr = np.arange(12, dtype=np.dtype("int32")).reshape(3, 4) * 2
+    var = Variable(data=arr, dims=["x", "y"])
+    ds = Dataset({"foo": var})
+    time.sleep(1)  # Make sure the checksum_date is at least one second in the future
+    ds.to_netcdf(netcdf_path)
+
+    # Now if we try to read the data back in, it should fail because the checksum_date
+    # is newer than the last_updated_at
+    with pytest.raises(IcechunkError):
```

Comment:
Upstream issue but `IcechunkError` seems like a unhelpfully-vague error type. I mean this assertion would pass if any other `IcechunkError` were raised for some unrelated reason...

Something more specific like `OutdatedVirtualReferenceError(IcechunkError)` seems like it might be more helpful?

Reply:
This will come with time, but for now this is what we get. Typing all the errors is a lot of work whihc we have on the roadmap.